### ISSUE TITLE
OZDEV - FILE protocol OZ device for ITS

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#2344]
"OZ" device for ITS by @cstacy, for accessing files on OZ through the FILE protocol.  Presumable there's a Twenex server somewhere.

**Comment by larsbrinkhoff** [comment]
Looks like it's using the LMFILE contact name though.

**Comment by ams** [comment]
There are two remote protocols for the Lisp Machine, LMFILE and REMOTE-FILE.   I don't know what MIT-OZ used.

**Comment by larsbrinkhoff** [comment]
This is between ITS and Twenex, so no Lispm involved.  ITS source says /LMFILE/.  Looking for Twenex code but slim pickings so far.

**Comment by larsbrinkhoff** [comment]
There is FCDEV for ITS to get files from the FileComputer: #321.

**Comment by cstacy** [comment]
I do not remember this at all!

(Beyond a vague recollection that maybe there was an OZ: device.)

**Comment by larsbrinkhoff** [comment]
That's ok, we can remember OZDEV for you wholesale.

**Comment by larsbrinkhoff** [comment]
There's a second independent OZDEV by @bawden: `title OZDEV -- foreign filesystem device.`  Earlier versions mentioned MIT-OZ, but the latest looks up the Chaosnet address for MIT-PREP.

**Comment by cstacy** [comment]
On 12/3/24 1:51 AM, Lars Brinkhoff wrote:

> That's ok, we can remember OZDEV for you wholesale.
>
I hope you can get my ass to OZ;  I may be overdrawn at the memory bank.


Now if you will excuse me, I need a Mokie-Cokey to calm down...

**Comment by cstacy** [comment]
It sounds like an experiment or WIP that was never completed, or 
something, which might be why I don't remember it.
Did you find this thing from an installed binary?
Or just happen across these source files while browsing?

**Comment by larsbrinkhoff** [comment]
I found it when searching for any kind of Twenex Chaosnet file server.  I don't see any binary, which is interesting.  If OZDEV were in use there ought to be a DEVICE;JOBDEV OZ, but there isn't.

I found this note:
```
Date: 02/09/83 11:49:46
From: CSTACY

Note: VAX device will be available same day as OZdevice.
```

ATSIGN JOBDEV does know about the OZ device.  The @DEV source has OZ being an MLDEV in versions 46 and 47.  52 and 53 make it an RMTDEV.  It's a Chaosnet protocol but it's own thing, not FILE/NFILE/LMFILE.

**Comment by bawden** [comment]
I seem to recall that the _original_ OZ: device was written by Gail Zacharias (@gzacharias, GZ on ITS).  That inspired a bunch of activity both to improve her code and to work on other remote filesystem devices.  You'll probably find a bunch of other detritus from that era if you look around...

**Comment by larsbrinkhoff** [comment]
Yes, RMTDEV was written by GZ.  The TITLE calls it "MLDEV for non-ITS hosts".  It has special cased code to handle devices KANSAS/OZKS and SP/SPEECH.  The contact name is `RMTDEV`.

Minor survey of other file servers, clients, and protocols follows.

The FILE server has conditionals to run on all of ITS, TOPS-20, and TENEX (maybe due to Symbolics' Foonly?).  The contact name is `FILE`.

@cstacy's OZDEV uses the contact name `LMFILE`, but I haven't found any such server for TOPS-20.

@bawden's OZDEV ... can't find a contact name, so looks like unfinished code.

The FCDEV client uses the contact name `LMFILE`, which makes sense for talking to the FC Lispm.

CFTP uses the contact name `FILE`.  Help text and comments mention OZ and Twenex, not so much Lispms.

@bawden's `NFILE` server for ITS was a latecomer in 1988.

I started making some notes about the timeline of file servers and protocols: https://github.com/PDP-10/its/issues/2346

**Comment by bictorv** [comment]
> Yes, RMTDEV was written by GZ. The TITLE calls it "MLDEV for non-ITS hosts". It has special cased code to handle devices KANSAS/OZKS and SP/SPEECH. The contact name is `RMTDEV`.

There is a RMTSLV.MID for TOPS-20, which is a server for the RMTDEV contact.

**Comment by larsbrinkhoff** [comment]
@bictorv, I'd be happy to test that, but I don't have any Chaos networked TOPS-20.  ITS already has RMTDEV installed, but it needs some links from DEVICE; the two-character device name is looked up as a network host.
